frontend/src/hooks/useActions.ts

```diff
@@ -674,4 +685,80 @@ export const selectionToCopyTemplate = (stateIds: number[], commentIds: number[]
   }
 }
 
+// Functions for importing and exporting modules
+export const useParseModuleFile = <T>(onData: (val: T) => void, errorMessage: string, input: File, onFinishLoading: () => void, onFailedLoading: () => void) => {
```

Comment:
These functions seem to just be copied from the existing project functions? Could've basically been reused no?